app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java

```diff
@@ -1,36 +1,39 @@
 package com.dimagi.biometric.activities;
 
-import android.os.Bundle;
+import com.dimagi.biometric.R;
 
 import org.commcare.commcaresupportlibrary.identity.IdentityResponseBuilder;
 import org.commcare.commcaresupportlibrary.identity.model.MatchResult;
 import org.commcare.commcaresupportlibrary.identity.model.MatchStrength;
 
+import java.util.ArrayList;
+
 import Tech5.OmniMatch.MatcherCommon;
 
 public class VerifyActivity extends BaseActivity {
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-
-        MatcherCommon.Record caseRecord = getRecordFromIntent();
-        if (caseRecord != null && caseId != null) {
+    protected void onCaptureSuccess(MatcherCommon.Record activeRecord) {
+        if (templateStr == null) {
+            return;
+        }
+        MatcherCommon.Record caseRecord = templateViewModel.getRecordFromTemplateStr(templateStr);
+        if (caseRecord != null) {
             templateViewModel.insertRecord(caseRecord, caseId);
         }
-    }
-
-    @Override
-    protected void onCaptureSuccess(MatcherCommon.Record activeRecord) {
         float score = templateViewModel.verifyRecord(activeRecord, caseId);
-        IdentityResponseBuilder.verificationResponse(
-                caseId, new MatchResult((int) (score * 100), getMatchStrength(score))
-        ).finalizeResponse(this);
+        finishWorkflow(score);
     }
 
     @Override
     protected void onCaptureCancelled() {
+        finishWorkflow(0f);
+    }
+
+    private void finishWorkflow(float score) {
```

Comment:
@zandre-eng can we rename `finishWorkflow` to something more specific?

Reply:
@avazirna It has been renamed to `finalizeResponse` ([f5b235c](https://github.com/dimagi/biometric-android/pull/7/commits/f5b235ceb0ce450dc481ce25bb4f8407cb2d83c5)).